### PR TITLE
Subject Access Request admin frontend

### PR DESCRIPTION
**Pull request by Calum-Campbell**
- Add SAR functionality to admin
- Only display to users that are added to the /identity/identity-admin/${STAGE}/dataExportUsers , configuration in identities parameter store. This is also enforced on the actual API request by requiring dataExportUserAuth on the UserSubjectAccessRequest .getById endpoint in case. 
- Button will start a download of a txt file and not change screen
- Related admin-api PR: https://github.com/guardian/identity-admin-api/pull/216

image

### CODE REVIEW

File: app/services/AdminApi.scala [hunk 1]
```diff
@@ -210,4 +210,15 @@ class AdminApi @Inject()(
       Left(CustomError("Fatal Error", contact))
     }
   }
+
+  def getUserSubjectAccessRequest(id: String) = {
+    requestSigner.sign(wsClient.url(s"$baseUrl/user/${id}/sar-request")).get().map {
+      case response if response.status == 200 => Right(response)
```
Comment (jfsoul) [inline app/services/AdminApi.scala]: all the other methods seem to use `checkResponse` for this

File: app/controllers/AccessUser.scala [hunk 2]
```diff
@@ -13,11 +12,10 @@ import util.Logging
 import scala.concurrent.ExecutionContext
 import scala.language.implicitConversions
 
-class AccessUser @Inject() (
-    googleAuth: GoogleAuth,
+class AccessUser @Inject()(
     cc: ControllerComponents,
     adminApi: AdminApi,
-    implicit val conf: Config)(implicit ec: ExecutionContext) extends AbstractController(cc) with Logging with I18nSupport {
+    implicit val conf: Config)(implicit ec: ExecutionContext, dataExportUserAuth: DataExportUserAuth, googleAuth: GoogleAuth) extends AbstractController(cc) with Logging with I18nSupport {
```
[the remark this answers is not in the document]
Comment (Calum-Campbell) [reply app/controllers/AccessUser.scala]: I made it implicit so I didn't have to inject it into every mention of `views.html.editUser()` (we need it in the template now so we can check wether to show the button or not)

File: app/services/AdminApi.scala (hunk 1 above)
Comment (Calum-Campbell) [reply app/services/AdminApi.scala]: Yeah didn't use that as the response is some sort of binary file, so seems to make more sense just to check for status==200

File: app/views/editUser.scala.html [hunk 3]
```diff
@@ -111,6 +114,16 @@
                         <input type="hidden" name="id" value="@userId" />
                         <input type="hidden" name="email" value="@userEmail" />
                     </form>
+                    @if(hasDataExportPermissions) {
+                        <form method="get" action="@routes.UserSubjectAccessRequest.getById">
```
[the remark this answers is not in the document]
Comment (Calum-Campbell) [reply app/views/editUser.scala.html]: Just used an <a> link now, thanks

File: app/services/AdminApi.scala [hunk 4]
```diff
@@ -210,4 +210,15 @@ class AdminApi @Inject()(
       Left(CustomError("Fatal Error", contact))
     }
   }
+
+  def getUserSubjectAccessRequest(id: String) = {
+    requestSigner.sign(wsClient.url(s"$baseUrl/user/${id}/sar-request")).get().map {
```
[the remark this answers is not in the document]
Comment (Calum-Campbell) [reply app/services/AdminApi.scala]: updated

File: app/controllers/AccessUser.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (QuarpT) [reply app/controllers/AccessUser.scala]: You could have an `@Inject` class that wraps editUser and takes `dataExportUserAuth` and `googleAuth` as injected parameters and passes a boolean ` dataExportUserAuth.isDataExportUser(googleAuth.userIdentity(request).map(_.email))` to the view

**Pull request merged**